assimilation_code/programs/filter/filter.rst

```diff
@@ -160,10 +160,13 @@ The detailed execution flow inside the filter program is:
    -  If requested, compute and write the "postassim" netcdf diagnostic files (members, mean, spread). This is BEFORE
       any posterior inflation has been applied.
    -  Apply posterior inflation if requested.
+   -  If requested, compute and write the "analysis" netcdf diagnostic files (members, mean, spread). 
+      This is AFTER any posterior inflation has been applied.
    -  Compute ensemble of posterior observation values with forward operators.
    -  Compute posterior observation space diagnostics.
-   -  If requested, compute and write out the "output" netcdf diagnostic files (members, mean, spread). This is AFTER
-      any posterior inflation has been applied.
+   -  If requested (for debugging) write observations available at this time to the observation sequence file.
```

Comment:
When I looked in filter_mod.f90 for the place where stage "assim" is written, 
I noticed that there's a section right after that place, within the AdvanceTime loop,
which can write _some_ obs to the obs_seq.final file, if the user turns on `write_obs_every_cycle`.
Then there's a section after the time loop which always writes the obs to obs_seq.final.
Maybe it should be

> If requested (usually for debugging) write the observations _which are in this time window_ to the observation sequence file.

It looks to me like setting `write_obs_every_cycle` could result in redundant writes of obs to obs_seq.final; in the AdvanceTime loop obs from each window would be written, 
then after the loop, all the obs would be written.  But that would be a separate issue from this.

Reply:
don't put it in if it is a debugging option.

Reply:
i agree since it's only for debugging and only for models which loop inside filter doing multiple assimilation windows, it's probably too much.  but for future reference, any mention of observation sequence files relative to filter needs to specify if it's the input obs_seq.out (which is a confusing name but historical) or the output obs_seq.final.

Reply:
I agree about obs_seq.{in,out,final}.
I'll remove the debugging item from the outline.